### PR TITLE
New sample: Edit feature-linked annotation

### PR DESCRIPTION
**Pull request by ZackAllen**

### CODE REVIEW

File: src/Android/Xamarin.Android/ArcGISRuntime.Xamarin.Samples.Android.csproj
Comment (nCastle1) [inline src/Android/Xamarin.Android/ArcGISRuntime.Xamarin.Samples.Android.csproj:585]: We'll want to be careful about changing this before merge.

File: src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.cs
Comment (nCastle1) [inline src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.cs:34]: I think for sample solution generator to work, there needs to be a layout file attribute IIRC

File: src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/readme.metadata.json
Comment (nCastle1) [inline src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/readme.metadata.json:17]: Script may need to be updated for 100.10

File: src/Forms/Android/ArcGISRuntime.Xamarin.Forms.Android.csproj
Comment (nCastle1) [inline src/Forms/Android/ArcGISRuntime.Xamarin.Forms.Android.csproj:106]: This will need to change before merge

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.xaml
Comment (nCastle1) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.xaml:30]: I think XAML files may need reformatting to follow .NET samples style. IIRC x:Name should be on the first line, and grid row/column/span values can be put on one line

**Review by nCastle1 — CHANGES_REQUESTED** [review]
I did a partial review just looking at the code and files, haven't had a chance to run yet. Looks really great so far, just a few things to be addressed before moving forward.

File: src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/readme.metadata.json
Comment (ZackAllen) [reply src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/readme.metadata.json:17]: I will do this in a separate PR. It will change a lot of files.

File: src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.cs
Comment (nCastle1) [inline src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.cs:68]: Typo, should be `parcelAnnotationLayer` (missing l)

File: src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.cs
Comment (nCastle1) [inline src/Android/Xamarin.Android/Samples/Data/EditFeatureLinkedAnnotation/EditFeatureLinkedAnnotation.cs:111]: I recommend either using `First()` if the expectation is that there will always be at least one Feature in the collection, or using `FirstOrDefault` and guarding against a null `_selectedFeature` value.

**Review by nCastle1 — CHANGES_REQUESTED** [review]
I think I'd like to see a few small changes to improve the interaction with the sample. I found it a little unintuitive to use as it is.

* I'd like to see an instruction label guiding through the process. E.g. '1. tap to select feature', '2. edit feature attributes', '3. tap again to move feature'
* It felt strange not being able to move the feature until after the feature editor was dismissed. I'd either like the ability to move the feature while the editor is opened, or very clear labeling about when I can and can't move the feature
* Should there be an option to not move the feature? Even after canceling I can't select another feature without moving the existing feature.

If any of that would substantially complicate the sample beyond what you think is reasonable, it is probably fine to leave mostly as-is; I will insist on some form of persistent help/status label.

**Review by nCastle1 — APPROVED** [review]
Approved pending resolution of two comments for iOS and any other remaining feedback.

**Pull request merged**
